test/iterio_test.py

```diff
@@ -101,3 +101,98 @@ def test_write_methods(self, buffer, method_name, method_args):
 
         with pytest.raises(io.UnsupportedOperation):
             method(*method_args)
+
+
+class TestIteratorBufferedIO:
```

Comment:
And since a lot of this is repetition, I would hope that condensing the base classes would mean we could get away with just a couple extra tests on the original class, to test just the added branches/cases.